File: delnms.cpp
#include<iostream>
#include<vector>
#include<string.h>
#include<stdio.h>
#include<map>
#include<math.h>
#include<algorithm>
#define LL long long
#define pb push_back
using namespace std;
vector<int>a;

main()
{
    vector<int>left,right;
    int N,i,j,flag,l,key,q,cd;
    scanf("%d",&N);
    a.pb(0);
    int x;
    int times[100001];
    memset(times,0,sizeof(times));
    for(i=1;i<=N;i++)
    {
        scanf("%d",&x);
        a.pb(x);
        times[x]++;
    }
    flag=1;
    for(i=a.size()-1;i>0 ;i--)
    {
        //cout<<"single element\n";
        if(times[ a[i]]==1)
        {
           // cout<<"singleton\n";
            times[a[i]]--;
            left.pb(i);
            right.pb(a.size()-1);
            a.erase(a.begin()+i);
        }
    }
    l = a.size();
   // cout<<" list size after single elements del ="<<l<<"\n";
    while(a.size()>1)
    {
        if(times[a[i]]==1)
        {
            times[a[i]]--;
            left.pb(i);
            right.pb(a.size()-1);
            a.erase(a.begin()+i);
        }
        else
        {
            key=a[a.size()-1];
            for(j=a.size()-2;j>0;j--)
            {
                if(a[j]==key)
                    break;
            }
            cd= a.size()-1-j;
            times[key]--;
            a.erase(a.begin()+(a.size()-1 ));

            while(j>0 && a[j]==key)
            {
                times[key]--;
                a.erase(a.begin()+j);
                j=j-cd;
            }
            j=j+cd; // one time cd extra added in the above while loop

            left.pb(j);
            right.pb(cd);
        }
    }

    q= left.size();
    cout<<q<<endl;
    for(i=0;i<q;i++)
    {
        printf("%d %d\n",left[i],right[i]);
    }



return 0;
}

File: delnms.py
n= int(input())
times=[]
for i in range(100001):
        times.append(0)
l= map(int,raw_input().split())
for i in range(l):
        times[l[i]]= times[l[i]]+1
        
right=[]
left=[]
while(len(l)>n-490):
        l_len= len(l)
        key=l[l_len-1]
        flag=1
        for i in range(l_len-2,-1,-1):
            #print("key=",key,"and ","li=",l[i])
            if(l[i]==key):
                diff=i
                flag=0
                left.append(i+1)
                right.append(l_len-1-i)
               # print("dual=",i+1,l_len-i-1)
                x=l.pop(l_len-1)
                y=l.pop(i)
                #print("popped= ",x,y)
                break
        if(flag==1):
            left.append(l_len)
            right.append(1)
            x=l.pop(l_len-1)
            #print("popped= ",x)
            #print(l_len,1)
            
            
n= len(right)
x=  len(l)
print(n+x)
for i in range(n):
    y=str(left[i])+" "+str(right[i])
    print(y)
for i in range(x,0,-1):
        y=str(i)+" 1"
        print(y)
                
                
            
